logger.go

```diff
@@ -0,0 +1,71 @@
+package gotbot
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Level is the level of the log generated.
+// It can be one of `Debug`, `Info`, `Warn`, or `Error`
+type Level string
+
+const (
+	Debug Level = "debug"
+	Info  Level = "info"
+	Warn  Level = "warn"
+	Error Level = "error"
+)
+
+// LogMessage contains the details of the log to be printed.
+type LogMessage struct {
+	Level Level     `json:"level"`
+	Msg   string    `json:"msg"`
+	Time  time.Time `json:"time"`
+}
+
+// Logger is used to print any messages that might be generated inside this package.
+// The default log is json formatted.
+type Logger interface {
+	Log(msg LogMessage)
```

Comment:
I think separating the interface method based on the log level would be a good idea because there will be some instances where I want to differentiate where the logs are streamed based on their level.
As an example that I might want the logs with info level to be logged to stdout while those with error level to stderr. Don't get me wrong I could implement it with this interface but the Log implementation would be bulky because I need to handle this additional condition.

Reply:
This will also remove the need to write helper function to generate the LogMessage.

Reply:
That seems a good idea too. What about the `LogMessage` type? if we have separate functions for each level the need for the type no longer exists. it can just be a regular string that is logged.

Reply:
How should we handle the `Time` field? Assuming the developer will take care of it seems naive.